lib/irb/cmd/edit.rb

```diff
@@ -37,7 +37,8 @@ def execute(*args)
               # in this case, we should just ignore the error
             end
 
-          if source
+          # Ignore `source.file == "(irb)"` and `source.first_line == nil` (binary file)
+          if source && File.exist?(source.file) && source.first_line
```

Comment:
I feel this should be encapsulated in a `Source#editable?` method.

Reply:
:+1:
Added `Source#file_exist?` and `Source#binary_file?`. `source.binary_file?` is also used from show_source